packages/wrangler/src/miniflare-cli/assets.ts

```diff
@@ -63,6 +68,71 @@ export default async function generateASSETSBinding(options: Options) {
 	};
 }
 
+/**
+ * An Undici custom Dispatcher that is used for the fetch requests
+ * of the Pages dev server proxy.
+ *
+ * Notably, the ProxyDispatcher reinstates the Host header that was removed by the
+ * Undici `fetch` function call. Undici removes the Host header as a security precaution,
+ * but this is not relevant for an internal Proxy.
+ *
+ * The ProxyDispatcher will delegate through to the current global `Dispatcher`,
+ * ensuring that the request is routed correctly in case a developer has changed the
+ * global Dispatcher by a call to `setGlobalDispatcher()`.
+ */
+class ProxyDispatcher extends Dispatcher {
+	private dispatcher: Dispatcher = getGlobalDispatcher();
+
+	constructor(private host: string | null) {
+		super();
+	}
+
+	dispatch(
+		options: Dispatcher.DispatchOptions,
+		handler: Dispatcher.DispatchHandlers
+	): boolean {
+		if (this.host !== null) {
+			this.reinstateHostHeader(options.headers, this.host);
+		}
+		return this.dispatcher.dispatch(options, handler);
+	}
+
+	close() {
+		return this.dispatcher.close();
+	}
+
+	destroy() {
+		return this.dispatcher.destroy();
+	}
+
+	/**
+	 * Ensure that the request contains a Host header, which would have been deleted
+	 * by the `fetch()` function before calling `dispatcher.dispatch()`.
+	 */
+	private reinstateHostHeader(
```

Comment:
It does not rely upon this, but I felt it was reasonable to keep it localised to this class (so not a free standing function), and keep it private (rather than a public class method).

Reply:
Oh my! I just found out that you can have private static class properties...